### DISCUSSION TITLE
Daily yield normalizied, but wrong data now

### DISCUSSION THREAD

**Discussion by lakatusch** [wlcrs/huawei_solar#449] [Setting up the integration]
hi there,

i had integrated my sun2000 inverter well, but i didn't want to see the discharge enery as the solar production.

So i used the code from this side: https://github.com/wlcrs/huawei_solar/wiki/Daily-Solar-Yield

and modified it to:

`  - name: "PV Huawei nur Solar Daily normalisiert"
    #friendly_name: "PV Energy daily normalisiert"
    unique_id: "PV Huawei nur Solar Daily normalisiert"
    device_class: energy
    state_class: total
    unit_of_measurement: kWh
    state: >-
      {% set energie = (states('sensor.inverter_daily_yield') | float) - (states('sensor.battery_day_discharge') | float) + (states('sensor.battery_day_charge') | float) %}
      {% if energie <= 0 %}
        {{ 0 }}
      {% else %}
        {{ energie }}
      {% endif %}`

i added the if clause, because i dont want to have negative values at night.

For the first day, it looks great:
![grafik](https://github.com/wlcrs/huawei_solar/assets/18687119/dc92666b-52b0-4647-a88f-e4a4488ce032)

but for the second day:
![grafik](https://github.com/wlcrs/huawei_solar/assets/18687119/1b9ba5bc-898c-4db6-a040-421852f0f486)

as you see, the day starts with the minus value from day before...

here the data:
![grafik](https://github.com/wlcrs/huawei_solar/assets/18687119/e5a6b611-fbd6-420c-ad0d-03482b802a2c)

any idea?

regards
Daniel

**Comment by Roving-Ronin** [comment]
@lakatusch To make it a bit easier for people to read, and make sure you don't have format issues, it would be a good idea to correctly insert you yaml into this thread, using the ``` on the line before and the line after your yaml code. This will allow github to recognise it as code and maintain your original format / line breaks etc.

**Reply by Gnec220** [reply]
Hi i have the same problem. were you able to solve it ? 
My Yaml code ist as below:

`template:
  - sensor:
      - name: energy_pv
        #friendly_name: "PV Energy"
        unique_id: "PV Energy"
        state: "{{ (states('sensor.inverter_total_yield') | float) - (states('sensor.battery_total_discharge') | float) + (states('sensor.battery_total_charge') | float)}}"
        device_class: energy      
        state_class: total
        unit_of_measurement: kWh

        #Corrected PV yield For Huawei with Battery    
#TDaily        
      - name: energy_pv_daily
        #friendly_name: "PV Energy daily"
        unique_id: "PV Energy daily"
        state: "{{ (states('sensor.inverter_daily_yield') | float) - (states('sensor.battery_day_discharge') | float) + (states('sensor.battery_day_charge') | float)}}"
        device_class: energy
        state_class: total
        unit_of_measurement: kWh
````

**Reply by Gnec220** [reply]
ok i solved it myself. i took the total energy as the solar production variable instead of daily variable.
